packages/vite/src/node/plugins/css.ts

```diff
@@ -556,16 +557,20 @@ export function cssPostPlugin(config: ResolvedConfig): Plugin {
           })
           chunk.viteMetadata.importedCss.add(this.getFileName(fileHandle))
         } else if (!config.build.ssr) {
-          // legacy build and inline css
-
-          // __VITE_ASSET__ and __VITE_PUBLIC_ASSET__ urls are processed by
-          // the vite:asset plugin, don't call resolveAssetUrlsInCss here
+          chunkCSS =
+            renderAssetUrlInJS(
+              this,
+              config,
+              chunk,
+              opts,
+              chunkCSS,
+              '`'
+            )?.toString() || chunkCSS
           chunkCSS = await finalizeCss(chunkCSS, true, config)
-
           const style = `__vite_style__`
           const injectCode =
             `var ${style} = document.createElement('style');` +
-            `${style}.innerHTML = ${JSON.stringify(chunkCSS)};` +
+            `${style}.innerHTML = \`${chunkCSS}\`;` +
```

Comment:
I think we are missing a test to detect a failure here. If we wrap `chunkCSS` in a template string, then the `"+new URL(...)+"` interpolation will not be executed